src/inventory/beacon_inventory.rs

```diff
@@ -0,0 +1,516 @@
+        }
+    };
+    match state {
+        Active => change_texture((0.0, 219.0, 22.0, 22.0)),
+        Pressed => change_texture((22.0, 219.0, 22.0, 22.0)),
+        Inactive => change_texture((44.0, 219.0, 22.0, 22.0)),
+        Focused => change_texture((66.0, 219.0, 22.0, 22.0)),
+    }
+}
+
+#[allow(dead_code)]
+#[derive(PartialEq)]
+enum ButtonState {
+    Active,
+    Pressed,
+    Inactive,
+    Focused,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[repr(u8)]
+#[allow(dead_code)]
+enum Effect {
+    Speed,
+    Slowness,
+    Haste,
+    MiningFatigue,
+    Strength,
+    InstantHealth,
+    InstantDamage,
+    JumpBoost,
+    Nausea,
+    Regeneration,
+    Resistance,
+    FireResistance,
+    WaterBreathing,
+    Invisibility,
+    Blindness,
+    NightVision,
+    Hunger,
+    Weakness,
+    Poison,
+    Wither,
+    HealthBoost,
+    Absorption,
+    Saturation,
+    Glowing,
+    Levitation,
+    Luck,
+    BadLuck,
+    SlowFalling,
+    ConduitPower,
+    DolphinsGrace,
+    BadOmen,
+    HerooftheVillage,
+    Darkness,
+}
+
+impl Effect {
+    fn get_texture(&self) -> String {
+        match self {
+            Effect::MiningFatigue => "minecraft:mob_effect/minig_fatigue".to_owned(),
+            Effect::InstantHealth => "minecraft:mob_effect/instant_health".to_owned(),
+            Effect::InstantDamage => "minecraft:mob_effect/instant_damage".to_owned(),
+            Effect::JumpBoost => "minecraft:mob_effect/jump_boost".to_owned(),
+            Effect::FireResistance => "minecraft:mob_effect/fire_resistance".to_owned(),
+            Effect::WaterBreathing => "minecraft:mob_effect/water_breathing".to_owned(),
+            Effect::NightVision => "minecraft:mob_effect/night_vision".to_owned(),
+            Effect::HealthBoost => "minecraft:mob_effect/health_boost".to_owned(),
+            Effect::BadLuck => "minecraft:mob_effect/bad_luck".to_owned(),
+            Effect::SlowFalling => "minecraft:mob_effect/slow_falling".to_owned(),
+            Effect::ConduitPower => "minecraft:mob_effect/conduit_power".to_owned(),
+            Effect::DolphinsGrace => "minecraft:mob_effect/dolphins_grace".to_owned(),
+            Effect::BadOmen => "minecraft:mob_effect/bad_omen".to_owned(),
+            Effect::HerooftheVillage => "minecraft:mob_effect/hero_of_the_village".to_owned(),
+            _ => format!("minecraft:mob_effect/{:?}", self).to_lowercase(),
+        }
+    }
+}
+
+impl TryFrom<i16> for Effect {
+    type Error = &'static str;
+    fn try_from(value: i16) -> Result<Self, Self::Error> {
+        if (0..34).contains(&value) {
+            Ok(unsafe { std::mem::transmute::<u8, Effect>(value as u8 - 1) })
+        } else if value == -1 {
+            Err("no effect")
+        } else {
+            Err("tried to set a effect that is unknown")
+        }
+    }
+}
+
+impl Into<u8> for Effect {
+    fn into(self) -> u8 {
+        unsafe { std::mem::transmute::<Effect, u8>(self) + 1 }
```

Comment:
i don't think we necessarily need a transmute here, but we could use an `as` cast if you specify the ordinal numbers for the enum values directly at their definition site like so: `Variant = 2,`